model/task/task.go

```diff
@@ -1632,3 +1635,62 @@ func (t *Task) GetJQL(searchProjects []string) string {
 
 	return fmt.Sprintf(jqlBFQuery, strings.Join(searchProjects, ", "), jqlClause)
 }
+
+// BlockedState returns "blocked," "pending," or "" to represent the state of the task
+// with respect to its dependencies
+func (t *Task) BlockedState() (string, error) {
+	if t.DisplayOnly {
+		return t.blockedStateForDisplayTask()
+	}
+
+	dependencyIDs := []string{}
+	for _, d := range t.DependsOn {
+		dependencyIDs = append(dependencyIDs, d.TaskId)
+	}
+	dependentTasks, err := Find(ByIds(dependencyIDs).WithFields(DisplayNameKey, StatusKey,
+		ActivatedKey, BuildVariantKey, DetailsKey, DependsOnKey))
+	if err != nil {
+		return "", errors.Wrap(err, "error finding dependencies")
+	}
+	taskMap := map[string]*Task{}
+	for _, t := range dependentTasks {
+		taskMap[t.Id] = &t
+	}
+	for _, dependency := range t.DependsOn {
+		depTask := taskMap[dependency.TaskId]
+		state, err := depTask.BlockedState()
+		if err != nil {
+			return "", err
+		}
+		if state == TaskBlocked {
+			return TaskBlocked, nil
+		} else if depTask.Status == evergreen.TaskSucceeded || depTask.Status == evergreen.TaskFailed {
+			if depTask.Status != dependency.Status && dependency.Status != AllStatuses {
+				return TaskBlocked, nil
+			}
```

Comment:
In the case where depTask.Status == evergreen.TaskSucceeded, and !(depTask.Status != dependency.Status && dependency.Status != AllStatuses), this function will return "", nil. Is this what you want?

Reply:
Yes, that means the dependency has been satisfied